src/writer/update-proposal.ts

```diff
@@ -0,0 +1,199 @@
+        `[writer] Failed to check proposal validation, ${space.id}, ${address}, ${JSON.stringify(
+          e
+        )}`
+      );
+      return 'failed to check validation';
+    }
+  }
+}
+
+export async function verify(body): Promise<any> {
+  const msg = jsonParse(body.msg);
+
+  const schemaIsValid = snapshot.utils.validateSchema(snapshot.schemas.updateProposal, msg.payload);
+  if (!schemaIsValid) {
+    log.warn('[writer] Wrong proposal format', schemaIsValid);
+    return Promise.reject('wrong proposal format');
+  }
+
+  const proposal = await getProposal(msg.space, msg.payload.proposal);
+  if (!proposal) return Promise.reject('unknown proposal');
+
+  const isChoicesValid = validateChoices({
+    type: msg.payload.type || proposal.type,
+    choices: msg.payload.choices.length > 0 ? msg.payload.choices : proposal.choices
+  });
+  if (!isChoicesValid) {
+    return Promise.reject(`wrong choices for "${msg.payload.type || proposal.type}" type voting`);
+  }
+
+  if (proposal.author !== body.address) return Promise.reject('Not the author');
+
+  const space = await getSpace(msg.space);
+  space.id = msg.space;
+
+  const spaceUpdateError = getSpaceUpdateError({
+    type: msg.payload.type || proposal.type,
+    space
+  });
+  if (spaceUpdateError) return Promise.reject(spaceUpdateError);
+
+  const hasScam = isScamDetected({
+    address: body.address,
+    name: msg.payload.name || proposal.title,
+    body: msg.payload.body || proposal.body
+  });
+  if (hasScam) return Promise.reject('scam proposal detected, contact support');
+
+  const authorizationError = await checkAuthorization(body.address, space);
+  if (authorizationError) return Promise.reject(authorizationError);
+
+  return Promise.resolve(proposal);
+}
+
+export async function action(body, ipfs, receipt, id, context): Promise<void> {
+  const originalProposal = context;
+  const msg = jsonParse(body.msg);
+  const updated = parseInt(msg.timestamp);
+  const metadata = msg.payload.metadata || {};
+  const plugins = JSON.stringify(metadata.plugins || {});
+
+  const proposal = {
+    ipfs,
+    updated,
+    type: msg.payload.type || originalProposal.type,
+    plugins: plugins || originalProposal.plugins,
+    title: msg.payload.name || originalProposal.title,
+    body: msg.payload.body,
+    discussion: msg.payload.discussion,
+    choices: JSON.stringify(msg.payload.choices) || originalProposal.choices
+  };
+
+  console.log('proposal', proposal, originalProposal);
+
+  const query = 'UPDATE proposals SET ? WHERE id = ?';
+  const params: any[] = [proposal, msg.payload.proposal];
+
+  await db.queryAsync(query, params);
+}
+
+export function getIpfsBody(body, context) {
+  const id = context.id;
+  const originIpfs = context.ipfs;
+  const { address, sig, ...restBody } = body;
+  const { message } = restBody.data;
+
+  const updatedProposal = {
+    ...context,
+    ...message
+  };
+
+  const ipfsBody = {
+    address,
+    sig,
+    hash: id,
+    originIpfs,
```

Comment:
We shouldn't change the format of the ipfs file

Reply:
do you propose to show only the data that was sent from UI to update a proposal?